pkg/gardener/shoot/extender/extensions/dns.go

```diff
@@ -74,9 +73,21 @@ func newDNSExtensionConfig(domain, secretName, dnsProviderType string) *DNSExten
 }
 
 func NewDNSExtension(shootName, secretName, domainSuffix, dnsProviderType string) (*gardener.Extension, error) {
-	domain := fmt.Sprintf("%s.%s", shootName, domainSuffix)
+	var providerConfig *DNSExtensionProviderConfig
+
+	if secretName != "" && dnsProviderType != "" && domainSuffix != "" {
+		domain := fmt.Sprintf("%s.%s", shootName, domainSuffix)
+		providerConfig = newDNSExtensionConfig(domain, secretName, dnsProviderType)
+	} else {
```

Comment:
Is this `else` statement still needed after you added similar check to `converter.go`?

```
	if opts.DNS.ProviderType != "" && opts.DNS.SecretName != "" && opts.DNS.DomainPrefix != "" {
		extendersForCreate = append(extendersForCreate, extender2.NewDNSExtender(opts.DNS.SecretName, opts.DNS.DomainPrefix, opts.DNS.ProviderType))
	}
```